datadotworld/client/api.py

```diff
@@ -441,6 +445,314 @@ def download_datapackage(self, dataset_key, dest_dir):
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> user_dataset = api_client.fetch_contributing_datasets()
+        {'count': 0, 'records': [], 'next_page_token': None}
+        """
+        try:
+            return self._user_api.fetch_contributing_datasets()
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    def fetch_liked_datasets(self):
+        """Fetch datasets that authenticated user likes
+
+        Parameters
+        ----------
+        limit : str, optional
+            Maximum number of items to include in a page of results
+        next : str, optional
+            Token from previous result page (to be used when requesting a subsequent page)
+        sort : str, optional
+            Property name to sort
+
+        Returns
+        -------
+        dict
+            Dataset definition, with all attributes
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> user_liked_dataset = api_client.fetch_liked_datasets()
+        >>> user_liked_dataset.count
+        1
+        """
+        try:
+            return self._user_api.fetch_liked_datasets()
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    def fetch_datasets(self):
+        """Fetch authenticated user owned datasets
+
+        Parameters
+        ----------
+        limit : str, optional
+            Maximum number of items to include in a page of results
+        next : str, optional
+            Token from previous result page (to be used when requesting a subsequent page)
+        sort : str, optional
+            Property name to sort
+
+        Returns
+        -------
+        dict
+            Dataset definition, with all attributes
+
+        Raises
+        ------
+        RestApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> import datadotworld as dw
+        >>> api_client = dw.api_client()
+        >>> user_owned_dataset = api_client.fetch_datasets()
+        >>> user_owned_dataset.records
+        []
+        """
+        try:
+            return self._user_api.fetch_datasets()
+        except _swagger.rest.ApiException as e:
+            raise RestApiError(cause=e)
+
+    # Sql Operations
+
+    def sql_get(self, dataset_key, query):
+        """Executes SQL queries against a dataset via GET
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+        query : str
+            SQL query
+        include_table_schema : bool
+            Flags indicating to include table schema in the response
+
```

Comment:
What does this function return? How can the user control that?

Reply:
`None` is not the right answer here. Anyone calling this method expects data back. The answer here is likely to be a file-like object that can be used in functions like `csv.reader`, `json.load` or `jsonlines.Reader`.